Sources/TuistLoader/Models+ManifestMappers/Target+ManifestMapper.swift

```diff
@@ -249,23 +252,44 @@ extension XcodeGraph.Target {
         var playgrounds: Set<AbsolutePath> = []
 
         // Sources
-        let allSources = try await XcodeGraph.Target.sources(
+        let globSources = try await XcodeGraph.Target.sources(
             targetName: targetName,
-            sources: manifest.sources?.globs.map { glob in
-                let globPath = try generatorPaths.resolve(path: glob.glob).pathString
-                let excluding: [String] = try glob.excluding.compactMap { try generatorPaths.resolve(path: $0).pathString }
-                let mappedCodeGen = glob.codeGen.map(XcodeGraph.FileCodeGen.from)
-                return XcodeGraph.SourceFileGlob(
-                    glob: globPath,
-                    excluding: excluding,
-                    compilerFlags: glob.compilerFlags,
-                    codeGen: mappedCodeGen,
-                    compilationCondition: glob.compilationCondition?.asGraphCondition
-                )
-            } ?? [],
+            sources: manifest.sources?.globs
+                .filter { $0.generationSource == nil }
+                .map { glob in
+                    let globPath = try generatorPaths.resolve(path: glob.glob).pathString
+                    let excluding: [String] = try glob.excluding.compactMap { try generatorPaths.resolve(path: $0).pathString }
+                    let mappedCodeGen = glob.codeGen.map(XcodeGraph.FileCodeGen.from)
+                    return XcodeGraph.SourceFileGlob(
+                        glob: globPath,
+                        excluding: excluding,
+                        compilerFlags: glob.compilerFlags,
+                        codeGen: mappedCodeGen,
+                        compilationCondition: glob.compilationCondition?.asGraphCondition
+                    )
+                } ?? [],
             fileSystem: fileSystem
         )
 
+        let scriptGeneratedSources = try manifest.sources?.globs
+            .filter { $0.generationSource == .targetScript }
+            .map { generated in
+                let pathString = try generatorPaths.resolve(path: generated.glob).pathString
+                let path = try AbsolutePath(validating: pathString)
+                if path.isGlobPath {
+                    throw TargetManifestMapperError.nonSpecificGeneratedResource(targetName: targetName, generatedSource: path)
+                }
+                let mappedCodeGen = generated.codeGen.map(XcodeGraph.FileCodeGen.from)
+                return XcodeGraph.SourceFile(
+                    path: path,
+                    compilerFlags: generated.compilerFlags,
+                    codeGen: mappedCodeGen,
+                    compilationCondition: generated.compilationCondition?.asGraphCondition
+                )
+            }
```

Comment:
I'd move this in the initial [map](https://github.com/tuist/tuist/pull/6990/files#diff-26a125b6de2c8c5bef2552bd263ab117d3aa62a6a6cd69c7e4ba045f48fecca1R257) and create appropriate helper functions instead if that `map` gets too complex.

Reply:
After trying - since they have a different return type (the former returns `SourceFileGlob` and the latter `SourceFile`, which is the type returned from `XcodeGraph.Target.sources` in the original map), attempting to share the code creates more code than it saves, so keeping as is for now. If this is not what you meant or you have a different suggestion - PLMK.

Reply:
I'd do the following instead:
```swift
sources: manifest.sources?.globs
   .map {
     switch $0 {
       case .alwaysPresent:
        return alwaysPresentSourceFiles(...)
      case .generated:
       return generatedSourceFile(...)
     }
   }
```

Reply:
Note that generated sources that are sent to `XcodeGraph.Target.sources(...)` method will eventually be filtered out (by `validateGlobPattern`), so calling `generatedSourceFile(...)` replaces the call to `XcodeGraph.Target.sources(...)`. There's still a possibility to do something like:
```
        manifest.sources?.globs.forEach { glob in
            if glob.type == .generated {
                generatedSource.append(glob)
            }
            else if glob.type == .alwaysPresent {
                presentSources.append(glob)
            }
        }
```
And then use each source array in the corresponding method

Reply:
Ahh, my bad, I understand the issue now – let's keep as-is